src/Arriba/Arriba/Diagnostics/Observability/IObservationRecorder.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+namespace Arriba.Diagnostics.Observability
+{
+    public interface IObservationRecorder : IDisposable
+    {
+        IObservationContext Context { get; }
+
+        void RecordTrace(string message);
+        void RecordEvent();
+        void RecordMetrics();
+        void RememberProperty(string name, string value);
+        void RememberMetric(string name, double value);
```

Comment:
I know that was in discussion already, but the methods naming was not easy for me to understanding what  to expect.

A Record<Something> for instance was expecting to pass something to be recorded, but it flushes the dictionary content. 

I think that it would be better to use. 
Record for the Remember methods and Flush for the record.

Reply:
Sure, this was also mentioned in the other PR, so I'll change them accordingly